src/braket/pennylane_plugin/translation.py

```diff
@@ -233,6 +286,13 @@ def translate_result_type(observable: Observable, targets: List[int]) -> ResultT
     if return_type is ObservableReturnTypes.Probability:
         return Probability(targets)
 
+    if return_type is ObservableReturnTypes.State:
+        if not targets and "StateVector" in supported_result_types:
+            return StateVector()
+        elif "DensityMatrix" in supported_result_types:
+            return DensityMatrix(targets)
+        raise NotImplementedError(f"Unsupported return type: {return_type}")
+
```

Comment:
It would be nice to convert this method to a `singledispatch` function, but that can be done post--launch.